lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -1844,13 +1845,21 @@ LogicalResult StmtEmitter::visitStmt(OutputOp op) {
   for (ModulePortInfo port : parent.getPorts()) {
     if (!port.isOutput())
       continue;
+
+    auto operand = op.getOperand(operandIndex);
+    if (operand.hasOneUse() && dyn_cast_or_null<InstanceOp>(operand.getDefiningOp())) {
+      ++operandIndex;
+      ++numStatementsEmitted;
+      continue;
+    }
+
     ops.clear();
     ops.insert(op);
     indent();
     if (isZeroBitType(port.type))
       os << "// Zero width: ";
     os << "assign " << emitter.outputNames[port.argNum] << " = ";
-    emitExpression(op.getOperand(operandIndex), ops);
+    emitExpression(operand, ops);
```

Comment:
Does your change still work if the port is of zero width? That could be a neat additional test case :+1:

Reply:
That's something I have not tested; I'll try it out now.

Reply:
I've just tested it, and it looks like the wires are correctly removed, but I don't think the generated Verilog is valid since the right parenthesis is in a comment.
```
hw.module.extern @B() -> (%B_output : i0)

hw.module @A() -> (%A_output : i0) {
    %0 = hw.instance "b1" @B() : () -> (i0)
    hw.output %0 : i0
}
```
```
// external module B

module A(
  // output /*Zero Width*/ A_output
   );

  B b1 (        // ../test.mlir:5:10
  //.B_output (A_output));
endmodule
```

Reply:
Cool! This might be an issue with the existing module emission. Might be a worthwhile follow-up PR on ExportVerilog to specifically fix this issue (unrelated to your nice fix here). I'd still recommend adding your test case for zero-width ports; you should be able to just do something like `// CHECK: //.B_output (A_output)` to not have the Verilog emission issue affect you :wink: